language/move-analyzer/src/symbols.rs

```diff
@@ -191,6 +212,125 @@ pub struct SymbolicatorRunner {
+                    s.push_str(type_list_to_ide_string(type_args).as_str());
+                    s.push('>');
+                    s
+                } else {
+                    "".to_string()
+                };
+                let acquires_str = if !acquires.is_empty() {
+                    let mut s = " acquires ".to_string();
+                    s.push_str(type_list_to_ide_string(acquires).as_str());
+                    s
+                } else {
+                    "".to_string()
+                };
+                let ret_str = match ret {
+                    sp!(_, Type_::Unit) => "".to_string(),
+                    _ => format!(": {}", type_to_ide_string(ret)),
+                };
+
+                write!(
+                    f,
+                    "fun {}::{}::{}{}({}){}{}",
+                    addr_to_ide_string(&mod_ident.address),
+                    mod_ident.module.value(),
+                    name,
+                    type_args_str,
+                    type_list_to_ide_string(args),
+                    ret_str,
+                    acquires_str
+                )
+            }
+        }
+    }
+}
+
+fn type_to_ide_string(sp!(_, t): &Type) -> String {
+    match t {
+        Type_::Unit => "()".to_string(),
+        Type_::Ref(mut_, s) => format!(
+            "&{} {}",
+            if *mut_ { "mut" } else { "" },
+            type_to_ide_string(s)
+        ),
+        Type_::Param(tp) => {
+            if tp.abilities.is_empty() {
+                format!("{}", tp.user_specified_name)
+            } else {
+                format!(
+                    "{}: {}",
+                    tp.user_specified_name,
+                    format_delim::<Ability, AbilitySetIntoIter>(
+                        tp.abilities.clone().into_iter(),
+                        " + "
+                    )
+                )
+            }
+        }
+        Type_::Apply(_, sp!(_, type_name), ss) => match type_name {
+            TypeName_::Multiple(_) => {
+                format!("({})", type_list_to_ide_string(ss))
+            }
+            TypeName_::Builtin(name) => {
+                if ss.is_empty() {
+                    format!("{}", name)
+                } else {
+                    format!("{}<{}>", name, type_list_to_ide_string(ss))
+                }
+            }
+            TypeName_::ModuleType(sp!(_, module_ident), struct_name) => {
+                let addr = addr_to_ide_string(&module_ident.address);
+                format!(
+                    "{}::{}::{}{}",
+                    addr,
+                    module_ident.module.value(),
+                    struct_name,
+                    if ss.is_empty() {
+                        "".to_string()
+                    } else {
+                        format!("<{}>", type_list_to_ide_string(ss))
+                    }
+                )
+            }
+        },
+        Type_::Anything => "_".to_string(),
+        _ => "invalid type".to_string(),
```

Comment:
Should you panic here? or something maybe indicate to the user its an internal error?

Reply:
And a nit, I try to avoid `_` patterns here. It would probably be helpful to list out which cases you don't expect

Reply:
> Should you panic here? or something maybe indicate to the user its an internal error?

The whole language server is implemented such that it avoids panicking as it would exit the server process. Feels like a better user experience to have some functionality missing (ideally being explicit about it) than dealing with the server crashes with the IDE does not handle well.

Reply:
I think at the very least, it should be a bit more clear to a dev that this is _not_ an intended type to have displayed

Reply:
> I think at the very least, it should be a bit more clear to a dev that this is _not_ an intended type to have displayed

I have changed it to:
```
        Type_::Var(_) => "invalid type (var)".to_string(),
        Type_::UnresolvedError => "invalid type (unresolved)".to_string(),
```